xds/src/test/java/io/grpc/xds/internal/security/certprovider/FileWatcherCertificateProviderProviderTest.java

```diff
@@ -106,6 +107,29 @@ public void createProvider_fullConfig() throws IOException {
             eq("/var/run/gke-spiffe/certs/certificates2.pem"),
             eq("/var/run/gke-spiffe/certs/private_key3.pem"),
             eq("/var/run/gke-spiffe/certs/ca_certificates4.pem"),
+            eq(null),
+            eq(7890L),
+            eq(mockService),
+            eq(timeProvider));
+  }
+
+  @Test
+  public void createProvider_spiffeConfig() throws IOException {
```

Comment:
Can we also test the case when the SPIFFE bundle is present and no CA certificates are present? IIUC, we decided yesterday that this was also a valid state (but not one that will be used initially).

Reply:
We're not going to change "minimum" number of fields (so the following test is still in place - https://github.com/grpc/grpc-java/blob/master/xds/src/test/java/io/grpc/xds/internal/security/certprovider/FileWatcherCertificateProviderProviderTest.java#L159)
With the new functionality if the `ca_certificate_file` is pointing to a non existing file then it _might_ be ok assuming spiffe trust map scenario can handle the verification

Reply:
Sorry I don't understand: why do we want to continue to require the `ca_certificate_file`?

In practice, I agree that both will often be set. However, shouldn't it be allowed in the future to have an xDS server that only returns the SPIFFE trust map if its clients know to use the SPIFFE trust map?

Reply:
IIRC, the conclusion of our conversation yesterday was that we should support the case when only the SPIFFE trust map is specified and there is no flat trust bundle - can we add a test for that case?

Reply:
Hi @ejona86 , we had an offline talk with Matt and agreed that there will be a point of time in future where control plane starts sending only SPIFFE trust maps but no flat trust bundles.
Question - is that ok to change the validation logic in this PR to make either `spiffe_trust_bundle_map_file` or `ca_certificate_file` mandatory for configs?

Reply:
No. The file watcher can be used for other purposes. And if we are already allowing those not to be set, then it would break users on upgrade if we start requiring it (unnecessarily).

Reply:
Oh, I see. `ca_certificate_file` is required today. I thought that was supposed to be optional. Then yes, that should likely become an "or". But the behavior should also be defined in the gRFC. I don't see the existing required behavior defined in the file_watcher Certificate Provider section of gRFC A29, although it could be deep in the provider discussion (I don't remember how a provider can say it doesn't have info).

This needs to be cross-language behavior, so it should be defined and this does whatever it is specified to do.

Reply:
Leaving temp link to that section of new gRFC - https://github.com/grpc/proposal/blob/266c1be0001cac009650e9198cc2e9a9376d6c01/A87-mtls-spiffe-support.md#configuring-spiffe-verification-via-xds